pkg/utils/pod/scheduling.go

```diff
@@ -137,6 +137,17 @@ func IsStuckTerminating(pod *corev1.Pod, clk clock.Clock) bool {
 	return IsTerminating(pod) && clk.Since(pod.DeletionTimestamp.Time) > time.Minute
 }
 
+// IsDrainable checks if a pod can be drained by Karpenter
+// It checks whether the following is true for the pod:
+// - Doesn't tolerate the "karpenter.sh/disruption=disrupting" taint
+// - Is not stuck terminating
+// - Isn't a mirror pod (https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/)
+func IsDrainable(pod *corev1.Pod, clk clock.Clock) bool {
+	return !ToleratesDisruptionNoScheduleTaint(pod) &&
```

Comment:
Do we want to modify `IsWaitingEviction` and put this function next to it since this is now a subset of that other function? So then `IsWaitingEviction` is just looking at the pods that exist that are drainable + removing the pods that are already terminated